src/compiler/scala/tools/nsc/typechecker/NamesDefaults.scala

```diff
@@ -27,7 +27,9 @@ trait NamesDefaults { self: Analyzer =>
   // we need the ClassDef. To create and enter the symbols into the companion
   // object, we need the templateNamer of that module class. These two are stored
   // as an attachment in the companion module symbol
-  class ConstructorDefaultsAttachment(val classWithDefault: ClassDef, var companionModuleClassNamer: Namer)
+  class ConstructorDefaultsAttachment(val classWithDefault: ClassDef, var companionModuleClassNamer: Namer) {
+    var defaults = mutable.ListBuffer[Symbol]()
```

Comment:
why is this a var?

Reply:
It should be a val, I guess I started out with a `var default: List[..]` but it worked out more convenient to use a mutable collection.